projectDocs/translating/crowdin.md

```diff
@@ -0,0 +1,153 @@
+
+- `%d`, digits
+- `%s`, string
+- `%.2f`, a number to 2 digits after the comma, for example 5.25
+- `{keyword}`, the text around the keyword should be translated, but the word should be left as is.
+
+#### Examples
+
+- `%d percent`: this means that `%d` will be replaced by a number when the program is running, and you only need to translate the word `percent`.
+- `subject: %s`: here `%s` means that another string will be substituted.
+- `{color} on {backgroundColor}`: In this case, the word `on` should be translated.
+The rest should be left alone or rearranged in order to suit the language.
+This will be presented as "black on white", "yellow on black", etc.
+
+### Messages with ampersands - shortcut keys
+
+Example: `&Rate`
+
+The letter following the ampersand sign can be used as a shortcut key.
+So when you translate, you can put the ampersand wherever in the translated message, and the letter that follows it will be used as the shortcut for your language.
+When you have completed all other translation work, you may want to review the shortcut keys, since they provide many users a fast way of jumping to particular items in dialogs, such as specific checkboxes or combo-boxes.
+It is best to try not to have duplicated keys.
+
+### Plural forms
+
+Languages can have any number of plural forms.
+Strings can have multiple versions depending on the plural form of the subject.
+
+Example:
+
+- `with %s item`
+- English plural form:
+  - Multiple: `with %s items`
+- Arabic plural forms:
+  - Zero: `تتضمن %s من العناصر`
+  - One: `تتضمّن %s من العناصر`
+  - Two: `تتضمَّن عنصرين %s`
+  - Few: `تتضمَّن %s عناصر`
+  - Many: `تتضمَّن %s من العناصر`
+  - Other: `تتضمَّن %s من العناصر`
+
+In the [Crowdin web interface](https://support.crowdin.com/online-editor/), these can be set for each language using the "Form" section which replaces the standard translation edit box.
+
+In PoEdit, the standard translation edit box has tabs for each plural form.
+[Object navigation](https://www.nvaccess.org/files/nvda/documentation/userGuide.html#ObjectNavigation) is required to move focus to each tab button for the plural form.
+You can do this by:
+
+- By moving to the previous focus object from the edit box, you can cycle through each plural form tab button by continuing to more backward.
+  - Desktop: `NVDA+numpad4`
+  - Laptop: `NVDA+shift+leftArrow`
+- Activate the current object once the desired tab button is reached.
+  - Desktop: `NVDA+numpadEnter`
+  - Laptop: `NVDA+enter`
+
+If the number of plural forms for your language is incorrect please message the [translators mailing list](https://groups.io/g/nvda-translations) or <info@nvaccess.org>.
+
+### String groupings
+
+Translators may wish to provide two different translations for the same English string depending on the context.
+Translation strings can be grouped by a tag to differentiate contexts.

```

Comment:
Wouldn't "with" be better here?

```suggestion
Translation strings can be grouped with a tag to differentiate contexts.
```

Reply:
I think they are equivalent, no?

Reply:
@coderabbitai which grammar is more correct here? "grouped by" or "grouped with"?

Reply:
interesting, learn some new thing every day huh

Reply:
The Code Rabbit AI wrote:
> @seanbudd In the context provided, "grouped by a tag" is more grammatically appropriate. This phrasing clearly indicates that the tag is the criterion used for grouping, which aligns well with the intention of differentiating translation strings based on specific contexts. "Grouped with a tag" could imply that the tag is merely accompanying the strings rather than serving as the basis for grouping.

@seanbudd, the rabbit is correct in explaining both sense but cannot guess what you wanted to write. Maybe it was neither clear to me what you wanted to write.

But "grouped with tag" is more representative of my personal experience. That is, the tag appears along with the string to translate to provide more context (as the translators comment also does).

I am not sure at all that all the strings with context "addonStore" are following each other in the list of strings to translate, making a group of strings related one with another (i.e. "grouped by tage"): at least, it is not the case when untranslated strings are grouped together, either in poedit or on Crowdin's web interface.

It's not a matter of grammar correctness but of meaning.

Reply:
I think you are interpreting grouping as being in the same physical group in all list views, as opposed to contextual grouping. These tags are logically grouped based on context, for example, you can search by tag in PoEdit and Crowdin.

Reply:
Yes, I was rather interpreting "grouping" as physical grouping rather than contextual grouping.

Anyway, my personal impression is that there is no specific grouping, i.e. "addonStore" strings are not gathered logically. I feel more that the context is an additional information / restriction indicating in which case this translation applies. To summarize, I would not use  "grouping" at all, but rather speak of tags and context only.

Though, it's really a matter of personal and subjective feeling. Feel free to ignore my comments on this topic, the more if no one expresses the same feeling!